src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private class Node {
+        private T element;
+        private Node next;
+        private Node prev;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        Node newNode = new Node(value);
+        if (index == size) {
+            addNodeWhenIndexEqualsSize(newNode);
+        } else {
+            Node findNode = findNode(index, "Can't add by index " + index);
+            insertNode(findNode, newNode);
+        }
+    }
+
+    private void addNodeWhenIndexEqualsSize(Node newNode) {
+        if (size == 0) {
+            tail = head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
```

Comment:
we actually have a method right for that logic which is called `add(T value)` :)
so let's move it to there at make them call each other and vice versa